src/context/cartContext.js

```diff
@@ -1,72 +1,81 @@
-            const response = await axios.post("/api/user/cart", { product: data }, {
-                headers: {
-                    authorization: userToken,
-                }
-            })
-            if (response.status === 201) {
-                loadingState(false)
-                setCart(response.data.cart);
-            }
-        } catch (error) {
-            console.log(error);
+  const [Cart, setCart] = useState([]);
+  async function addToCart(data, loadingState) {
+      const userToken = useLocalStorageGetItem("user-token");
+    try {
+      loadingState(true);
+      const response = await axios.post(
+        "/api/user/cart",
+        { product: data },
+        {
+          headers: {
+            authorization: userToken,
+          },
         }
-
+      );
+      if (response.status === 201) {
+        loadingState(false);
+        setCart(response.data.cart);
+      }
+    } catch (error) {
+      console.log(error);
     }
+  }
 
-    async function changeCartQuantity(quantityType, productID) {
-        
-        try {
-            const response = await axios.post(`/api/user/cart/${productID}`, {
-                action: {
-                    type: quantityType,
-                }
-            }, {
-                headers: {
-                    authorization: userToken
-                }
-            })
-            if (response.status === 200) {
-                setCart(response.data.cart);
-            }
-
-        } catch (error) {
-            console.log(error)
+  async function changeCartQuantity(quantityType, productID) {
+    const userToken = useLocalStorageGetItem("user-token");
+    try {
+      const response = await axios.post(
+        `/api/user/cart/${productID}`,
+        {
+          action: {
+            type: quantityType,
+          },
+        },
+        {
+          headers: {
+            authorization: userToken,
+          },
         }
+      );
+      if (response.status === 200) {
+        setCart(response.data.cart);
+      }
+    } catch (error) {
+      console.log(error);
     }
+  }
 
-    async function removeFromCart(productID) {
-        try {
-            const response = await axios.delete(`/api/user/cart/${productID}`,
-                {
-                    headers: {
-                        authorization: userToken
-                    }
-                })
-            if (response.status === 200) {
-                setCart(response.data.cart);
-            }
-        } catch (error) {
-            console.log(error);
-        }
+  async function removeFromCart(productID) {
+    const userToken = useLocalStorageGetItem("user-token");
+    try {
+      const response = await axios.delete(`/api/user/cart/${productID}`, {
+        headers: {
+          authorization: userToken,
+        },
+      });
+      if (response.status === 200) {
+        setCart(response.data.cart);
+      }
+    } catch (error) {
+      console.log(error);
     }
+  }
```

Comment:
suggestion - you can move these functions to a new file to make it more readable

Reply:
Then I have to make 2 functions as here I am able to use context but in functions, I have to send as a argument. Thats why chose not to make new file